src/databricks/labs/remorph/reconcile/connectors/snowflake.py

```diff
@@ -33,36 +50,28 @@ def read_data(self, catalog: str, schema: str, query: str, options: JdbcReaderOp
                     .load()
                 )
             return df.select([col(column).alias(column.lower()) for column in df.columns])
-        except PySparkException as e:
-            error_msg = (
-                f"An error occurred while fetching Snowflake Data using the following {query} in "
-                f"SnowflakeDataSource : {e!s}"
-            )
-            raise PySparkException(error_msg) from e
+        except RuntimeError as e:
+            raise self._raise_runtime_exception(e, "data", table_query)
 
     def get_schema(self, catalog: str, schema: str, table: str) -> list[Schema]:
+        schema_query = self.get_schema_query(catalog, schema, table)
         try:
-            schema_query = self.get_schema_query(catalog, schema, table)
             schema_df = self.reader(schema_query).load()
             return [Schema(field.column_name.lower(), field.data_type.lower()) for field in schema_df.collect()]
-        except PySparkException as e:
-            error_msg = (
-                f"An error occurred while fetching Snowflake Schema using the following {table} in "
-                f"SnowflakeDataSource: {e!s}"
-            )
-            raise PySparkException(error_msg) from e
+        except RuntimeError as e:
+            raise self._raise_runtime_exception(e, "schema", schema_query)
 
     def reader(self, query: str) -> DataFrame:
         options = {
-            "sfUrl": self._get_secrets('sfUrl'),
-            "sfUser": self._get_secrets('sfUser'),
-            "sfPassword": self._get_secrets('sfPassword'),
-            "sfDatabase": self._get_secrets('sfDatabase'),
-            "sfSchema": self._get_secrets('sfSchema'),
-            "sfWarehouse": self._get_secrets('sfWarehouse'),
-            "sfRole": self._get_secrets('sfRole'),
+            "sfUrl": self._get_secret('sfUrl'),
+            "sfUser": self._get_secret('sfUser'),
+            "sfPassword": self._get_secret('sfPassword'),
+            "sfDatabase": self._get_secret('sfDatabase'),
+            "sfSchema": self._get_secret('sfSchema'),
+            "sfWarehouse": self._get_secret('sfWarehouse'),
+            "sfRole": self._get_secret('sfRole'),
         }
-        return self.spark.read.format("snowflake").option("dbtable", f"({query}) as tmp").options(**options).load()
+        return self._spark.read.format("snowflake").option("dbtable", f"({query}) as tmp").options(**options).load()
```

Comment:
`reader` should just return the `DataFrameReader`.

Reply:
its already done.